rosidl_runtime_py/rosidl_runtime_py/set_message.py

```diff
@@ -34,7 +36,11 @@ def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
     for field_name, field_value in values.items():
         field_type = type(getattr(msg, field_name))
         try:
-            value = field_type(field_value)
+            if field_type == array.array:
+                field_typecode = getattr(msg, field_name).typecode
+                value = field_type(field_typecode, field_value)
+            else:
+                value = field_type(field_value)
         except TypeError:
             value = field_type()
```

Comment:
If we ever hit a `TypeError` when the `field_type` is `array.array`, then we'll throw a different exception here since you can't instantiate an `array.array` without a typecode.  However, I couldn't see how that would happen, so I didn't make any changes here.

Reply:
Please apply the same conditional logic in the exception case to construct and empty array.

Reply:
Fixed in a29f80f